qldbdriver/qldbdriver.go

```diff
@@ -79,14 +79,6 @@ func New(ledgerName string, qldbSession *qldbsession.QLDBSession, fns ...func(*D
 	logger := &qldbLogger{options.Logger, options.LoggerVerbosity}
 
 	driverQldbSession := *qldbSession
-	if qldbSession.Client != nil {
-		client := *qldbSession.Client
-
-		qldbSDKRetries := 0
-		client.Config.MaxRetries = &qldbSDKRetries
-
-		driverQldbSession.Client = &client
-	}
```

Comment:
Should we remove the protection of underlying value mutation of client? Also, seems the sdk retries need to be set to 0 explicitly. https://github.com/awslabs/amazon-qldb-driver-go/pull/54

Reply:
It's immutable since we copied the value by `driverQldbSession := *qldbSession`, the retry is disabled in https://github.com/awslabs/amazon-qldb-driver-go/blob/3d0d355100a8056f871ebee8706e0c70f316459c/qldbdriver/communicator.go#L48

Reply:
Is it just assigning the pointer instead of copying value?

I think we just need to make sure [client.options](https://github.com/aws/aws-sdk-go-v2/blob/service/qldbsession/v1.4.2/service/qldbsession/api_client.go#L26) is not immutable after the driver is created since we pass in a pointer of client when creating a new driver.

Reply:
Everything in Go is passed by value https://golang.org/doc/faq#pass_by_value, after the driver creation user could not mutate the `qldbSession` client. The options is not immutable as we need to override client options in `SendCommand` operations calls https://aws.github.io/aws-sdk-go-v2/docs/making-requests/#OverrideClientOptionsForOperation

Reply:
Gotcha. I mistakenly thought driverQldbSession is a pointer too.